pkg/controllerutils/seedfilter.go

```diff
@@ -82,9 +79,16 @@ func ShootIsManagedByThisGardenlet(shoot *gardencorev1beta1.Shoot, gc *config.Ga
 	return shoot.Spec.SeedName != nil && *shoot.Spec.SeedName == seedName
 }
 
-// ShootIsBeingMigratedToThisGardenlet checks if the given shoot is currently being migrated to this gardenlet.
-func ShootIsBeingMigratedToThisGardenlet(shoot *gardencorev1beta1.Shoot, gc *config.GardenletConfiguration) bool {
-	return ShootIsManagedByThisGardenlet(shoot, gc) && shoot.Status.SeedName != nil && *shoot.Spec.SeedName != *shoot.Status.SeedName
+// ShootIsBeingMigratedToSeed checks if the given shoot is currently being migrated to the seed with the given name.
+func ShootIsBeingMigratedToSeed(ctx context.Context, c client.Reader, shoot *gardencorev1beta1.Shoot, seedName string) bool {
```

Comment:
This function mixes things and definitely does more than the name suggests.
I'm confused by the usages that rely on this function to check the seed setting as well now?

Reply:
What exactly is confusing? For the force restore mechanism to work, both source and destination seeds should have `ownerChecks` enabled. The check of the destination seed happens when setting up the informer and workers in the controller. The check of the source seed should happen exactly here in my opinion - if the setting is not enabled, we are not even interested in this shoot and should not get events about it, and if by any chance we get one, the controller should not process it. Therefore both the filter and the controller are calling this function. Do you see a better way to achieve this?

Regarding the name, I think it's still rather accurate. I would add the above explanation to the doc string but I would propose not to change the name, as it's accurate enough and already long enough as well.

Reply:
Note that if the controllers are moved to GCM, this function should check both the source and the destination seeds.

Reply:
Ah, I was mostly confused by the naming and that the doc string didn't mention anything about the seed setting. 
With the amended doc string, this is clearer now and I'm fine with keeping the name.

Can you add the same comment to `BackupEntryIsBeingMigratedToSeed`?

Reply:
Done